dali/pipeline/operators/reader/loader/video_loader.h

```diff
@@ -117,6 +117,9 @@ class VideoLoader : public Loader<GPUBackend, SequenceWrapper> {
     /* Required to use libavformat: Initialize libavformat and register all
      * the muxers, demuxers and protocols.
      */
+
+    CUDA_CALL(cudaGetDevice(&device_id_));
```

Comment:
Why we need to call it now?

Reply:
It is needed by `NvDecoder`'s for the CUDA driver API, here:
https://github.com/NVIDIA/DALI/blob/a9603ae816dfabd42cac2083836e618f97b44aff/dali/pipeline/operators/reader/nvdecoder/nvdecoder.cc#L102
Since we instantiate one `NvDecoder` per container, it is good to have it at a `VideoLoader` level. Also it's good to have it for error messages.